base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -593,15 +593,50 @@ public boolean hasRevocationAuthorization(
             String accountID, Date time, ACMEIdentifier identifier)
             throws Exception {
 
-        // TODO: handle wildcard
+        boolean wildcard = false;
+        String ident = identifier.getValue();
+        if (IDENTIFIER_TYPE_DNS.equals(identifier.getType()) && ident.startsWith("*.")) {
+            wildcard = true;
+            ident = ident.substring(2);  // strip wildcard
+        }
 
+        /* RFC 8555 is unclear about whether wildcard revocations
+         * authorise non-wildcard identifiers, or vice-versa.
+         *
+         * This implementation takes the view that:
+         *
+         * AUTHZ-ID    WILDCARD   ID-TO-REVOKE  AUTHORISED?
+         *
+         * foo.example.com  Y    *.foo.example.com  Y
+         * foo.example.com  Y  bar.foo.example.com  N
+         * foo.example.com  Y      foo.example.com  Y
```

Comment:
Per above discussion, if we're doing strict wildcard matching, wouldn't this be a N in the authorized column as well? I understand the actual DNS challenge identifier is the same, but... ~shrugs~

Reply:
Yeah, this was a quirk of the LDAP schema (the query would have been really complex to avoid it).  @edewata suggested a tweak to the schema that will make it so we can more easily achieve the strict matching behaviour.  I will adopt that change.  So in the meantime, this PR can remain WIP.

Reply:
PR updated with schema changes, and strict identifier matching.